xml/System.IO/Directory.xml

```diff
@@ -2405,7 +2405,7 @@ An I/O error occurred.</exception>
           <paramref name="path" /> is <see langword="null" />.</exception>
         <exception cref="T:System.IO.PathTooLongException">The specified path, file name, or both exceed the system-defined maximum length.</exception>
         <exception cref="T:System.IO.IOException">
-          <paramref name="path" /> is a file name.</exception>
+          <paramref name="path" /> is a file name or when file or directory is corrupted and unreadable (example: invalid first allocation unit of a FAT32 partition).</exception>
```

Comment:
The "path is a file name" exception is unrelated to the new exception, that's why it should use the `-or-`. 
```xml
        <exception cref="T:System.IO.IOException"><paramref name="path" /> is a file name.
          
 -or- 
 
File or directory is corrupted and unreadable (example: invalid first allocation unit of a FAT32 partition).</exception>
```